src/Cake.Hosts/HostsFileUtil.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Cake.Hosts
+{
+    /// <summary>
+    /// Validate and extract data from hosts file
+    /// </summary>
+    public class HostsFileUtil
+    {
+        /// <summary>
+        /// Verify if line is from hosts file
+        /// </summary>
+        /// <param name="line">The line from hosts file</param>
+        /// <returns>If is not empty and not starts with #, is valid</returns>
+        static public bool IsLineAHostFile(string line) 
+            => line.Trim().Length > 0 && !line.TrimStart().StartsWith("#");
+
+        /// <summary>
+        /// Convert line in object HostsFile
+        /// </summary>
+        /// <param name="line">The line from hosts file</param>
+        /// <returns>If line is valid, return a object HostsFile</returns>
+        static public HostsFile GetHostsFile(string line)
+        {
+            var result = TryGetHostsFileEntry(line);
+
+            if (result == null)
+                throw new InvalidDataException();
```

Comment:
I'm not sure about this one. I'd rather not throw any exceptions if format of the string does not match, just skip that line with a possible log entry. 

Also, I feel there is no need to go 2 levels of methods that should be internal anyway. i.e. `TryGetHostsFileEntry` is not used anywhere outside of `GetHostsFile` and it does not add to the readability.